app/code/Magento/Customer/Model/Address.php

```diff
@@ -171,13 +171,14 @@ public function updateData(AddressInterface $address)
     public function getDataModel($defaultBillingAddressId = null, $defaultShippingAddressId = null)
     {
         if ($this->getCustomerId() || $this->getParentId()) {
-            if ($this->getCustomer()->getDefaultBillingAddress()) {
+            if ((intval($defaultBillingAddressId) == 0) && $this->getCustomer()->getDefaultBillingAddress()) {
```

Comment:
There is dedicated function is_null in php, but in Magento we use strict comparison equation (===) with null
So, in your case code should look like
`if ($defaultBillingAddressId === null) && $this->getCustomer()->getDefaultBillingAddress()) {`